### PR TITLE
RNAsum-0.4.1

### PR DESCRIPTION
**Pull request by skanwal**
Created, tested and registered CWL tool for RNAsums' latest container (0.4.1) with Dragen-WTS changes

### CODE REVIEW

File: tools/rnasum/0.4.1/rnasum__0.4.1.cwl [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+# ILMN Resources Guide: https://support-docs.illumina.com/SW/ICA/Content/SW/ICA/RequestResources.htm
+hints:
+  ResourceRequirement:
+    ilmn-tes:resources:
+      tier: standard
+      type: standardHiCpu
+      size: large
+  DockerRequirement:
+    dockerPull: "umccr/rnasum:0.4.1"
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/rnasum/0.4.1/rnasum__0.4.1.cwl:49]: Yes - that's on my to-do.
Can discuss in our next chat.

File: config/project.yaml [hunk 2]
```diff
@@ -306,6 +306,17 @@ projects:
             ica_workflow_version_name: 1.13.0
             modification_time: 2021-10-27T23:59:18UTC
             run_instances: []
+      - name: rnasum
```
[the remark this answers is not in the document]
Comment (skanwal) [reply config/project.yaml:309]: Don't think I can use `$ cwl-ica register-tool-run-instance` to point to task run in `dev`?

File: tools/rnasum/0.4.1/rnasum__0.4.1.cwl [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+          #!/usr/bin/bash
+
+          # Fail on non-zero exit code
+          set -euo pipefail
+
+          # Set work directory
+          cd /rmd_files
+
+          # Run rnasum with input parameters
+          eval Rscript ./RNAseq_report.R '"\${@}"'
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/rnasum/0.4.1/rnasum__0.4.1.cwl]: Updated in https://github.com/umccr/cwl-ica/commit/e3fb4d2da1ee0defff351b1ffec9a8d110fa5907

File: config/project.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (skanwal) [reply config/project.yaml:309]: Figured and updated in https://github.com/umccr/cwl-ica/commit/0d350f7af5f3bd38d4750ecb759ca9c4033d5a8a

File: tools/rnasum/0.4.1/rnasum__0.4.1.cwl (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexiswl) [reply tools/rnasum/0.4.1/rnasum__0.4.1.cwl:64]: It doesn't seem to exist, can you confirm you've pushed it.

image

[the remark this answers is not in the document]
Comment (skanwal) [reply tools/rnasum/0.4.1/rnasum__0.4.1.cwl:64]: I see the issue - it was apparently "private". I have made it public and testing again.

[the remark this answers is not in the document]
Comment (skanwal) [reply tools/rnasum/0.4.1/rnasum__0.4.1.cwl]: Moved to [quay](https://github.com/umccr/cwl-ica/commit/c6e85a19e0e63f44614215768ea3ab70a200329c) and tested `wfr.6ac9fd6da917415f95c6037d45c6c79c`

**Review by alexiswl — APPROVED** [review]

**Comment by alexiswl** [comment]
@skanwal not sure why the check failed, re-basing and manually triggered re-run of sync-workflow

**Comment by alexiswl** [comment]
Just looking at the graph from the catalogue and the summary from https://github.com/umccr/rnasum#2-wts-and-wgs-data, there seems to be a bit of a mismatch, standard hicpu from [the docs](https://support-docs.illumina.com/SW/ICA/Content/SW/ICA/RequestResources.htm) has 71 CPU and 140 Mem but docs say 1 CPU and 20 Gb

**Comment by alexiswl** [comment]
Searching https://github.com/umccr/cwl-ica/search?l=Common+Workflow+Language&q=standardhicpu there are 6 workflows using standardhicpu large instances.  I think the umccrise and bclconvert are justified.  
From the [arriba docs](https://arriba.readthedocs.io/en/latest/current-limitations/), both arriba fusion and arriba drawing could probably be reduced to a smaller instance type. This should mean that they're launched quicker too

**Comment by skanwal** [comment]
> both arriba fusion and arriba drawing could probably be reduced to a smaller instance type
That can be handled in a separate PR.

For RNAsum, we can probably also use standardHiCpu-medium but for now we can see how it goes with large instance and update if we see issues.

**Pull request merged**